x-pack/legacy/plugins/actions/server/builtin_action_types/webhook.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+import { nullableType } from './lib/nullable';
+import { ActionType, ActionTypeExecutorOptions, ActionTypeExecutorResult } from '../types';
+import { assertValidInterval } from '../../../task_manager/lib/intervals';
+
+// Generic Schemas
+const PORT_MAX = 256 * 256 - 1;
+const PortSchema = schema.number({ min: 1, max: PORT_MAX });
+
+// config definition
+export type ActionTypeConfigType = TypeOf<typeof ConfigSchema>;
+
+const HeadersSchema = schema.recordOf(schema.string(), schema.string());
+const ProxySchema = schema.object({
+  host: schema.string(),
+  port: nullableType(PortSchema),
+});
+
+const ConfigSchema = schema.object(
+  {
+    host: schema.string(),
+    port: PortSchema,
+    scheme: schema.oneOf([schema.literal('http'), schema.literal('https')], {
+      defaultValue: 'http',
+    }),
+    method: schema.oneOf(
+      [
+        schema.literal('head'),
+        schema.literal('get'),
+        schema.literal('post'),
+        schema.literal('put'),
+        schema.literal('delete'),
+      ],
+      {
+        defaultValue: 'get',
+      }
+    ),
+    path: nullableType(schema.string()),
+    url: nullableType(schema.string()),
+    headers: nullableType(HeadersSchema),
+    proxy: nullableType(ProxySchema),
+    connection_timeout: nullableType(schema.string()),
+    read_timeout: nullableType(schema.string()),
+  },
+  { validate: validateConfig }
+);
+
+function validateConfig(configObject: any): string | void {
+  const config: ActionTypeConfigType = configObject;
+
+  const { read_timeout: readTimeout, connection_timeout: connectionTimeout } = config;
```

Comment:
timeout is another one of these kinda general things, like http proxy, that I feel like we need to somehow expose higher, for all actions; and then actions could make use of it somehow, by passing that value into http calls as the timeout.  There's a card open for this: https://github.com/elastic/kibana/projects/26#card-24087404

Reply:
Do we expect all actions to be HTTP style actions?
We don't want to end up with Proxy settings for some kind of more "internal" action that interacts with Kibana for example.